File: Minesweeper/Minesweeper.py
import random

class Cell:
    def __init__(self):
        self.is_bomb = False
        self.revealed = False
        self.adjacent_bombs = 0

class Minesweeper:
    def __init__(self, size=5, bombs=5):
        self.size = size
        self.board = [[Cell() for _ in range(size)] for _ in range(size)]
        self.bombs = bombs
        self.place_bombs()
        self.calculate_adjacent()

    def place_bombs(self):
        count = 0
        while count < self.bombs:
            x, y = random.randint(0, self.size-1), random.randint(0, self.size-1)
            if not self.board[x][y].is_bomb:
                self.board[x][y].is_bomb = True
                count += 1

    def calculate_adjacent(self):
        for x in range(self.size):
            for y in range(self.size):
                if self.board[x][y].is_bomb:
                    continue
                for dx in [-1, 0, 1]:
                    for dy in [-1, 0, 1]:
                        nx, ny = x+dx, y+dy
                        if 0 <= nx < self.size and 0 <= ny < self.size and self.board[nx][ny].is_bomb:
                            self.board[x][y].adjacent_bombs += 1

    def reveal(self, x, y):
        if not (0 <= x < self.size and 0 <= y < self.size): return
        cell = self.board[x][y]
        if cell.revealed: return
        cell.revealed = True
        if cell.adjacent_bombs == 0 and not cell.is_bomb:
            for dx in [-1, 0, 1]:
                for dy in [-1, 0, 1]:
                    if dx != 0 or dy != 0:
                        self.reveal(x+dx, y+dy)

    def display(self):
        for row in self.board:
            print(' '.join('B' if cell.is_bomb and cell.revealed else
                           str(cell.adjacent_bombs) if cell.revealed else '*' for cell in row))

# Run Game
game = Minesweeper()
while True:
    game.display()
    x, y = map(int, input("Enter coordinates (x y): ").split())
    if game.board[x][y].is_bomb:
        print("Game Over!")
        game.board[x][y].revealed = True
        game.display()
        break
    game.reveal(x, y)

File: Minesweeper/Minesweeper.js
const readline = require('readline');

const SIZE = 5;
const BOMBS = 5;

let board = Array.from({ length: SIZE }, () => Array(SIZE).fill('0'));
let display = Array.from({ length: SIZE }, () => Array(SIZE).fill('*'));

function initBoard() {
    for (let i = 0; i < SIZE; i++) {
        for (let j = 0; j < SIZE; j++) {
            board[i][j] = '0';
        }
    }
}

function placeBombs() {
    let count = 0;
    while (count < BOMBS) {
        let x = Math.floor(Math.random() * SIZE);
        let y = Math.floor(Math.random() * SIZE);
        if (board[x][y] !== 'B') {
            board[x][y] = 'B';
            count++;
        }
    }
}

const dx = [-1, -1, -1, 0, 0, 1, 1, 1];
const dy = [-1, 0, 1, -1, 1, -1, 0, 1];

function countAdjacentBombs(x, y) {
    let count = 0;
    for (let d = 0; d < 8; d++) {
        let nx = x + dx[d];
        let ny = y + dy[d];
        if (nx >= 0 && nx < SIZE && ny >= 0 && ny < SIZE && board[nx][ny] === 'B') {
            count++;
        }
    }
    return count;
}

function calculateNumbers() {
    for (let i = 0; i < SIZE; i++) {
        for (let j = 0; j < SIZE; j++) {
            if (board[i][j] !== 'B') {
                board[i][j] = countAdjacentBombs(i, j).toString();
            }
        }
    }
}

function showBoard() {
    for (let i = 0; i < SIZE; i++) {
        let row = '';
        for (let j = 0; j < SIZE; j++) {
            row += display[i][j] + ' ';
        }
        console.log(row);
    }
}

function startGame() {
    initBoard();
    placeBombs();
    calculateNumbers();
    for (let i = 0; i < SIZE; i++) {
        for (let j = 0; j < SIZE; j++) {
            display[i][j] = '*';
        }
    }
    let revealed = 0;
    const totalToReveal = SIZE * SIZE - BOMBS;
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    function ask() {
        showBoard();
        rl.question('Enter x y: ', (answer) => {
            let [x, y] = answer.split(' ').map(Number);
            if (
                isNaN(x) || isNaN(y) ||
                x < 0 || x >= SIZE || y < 0 || y >= SIZE
            ) {
                console.log('Invalid coordinates. Try again.');
                ask();
                return;
            }
            if (display[x][y] !== '*') {
                console.log('Cell already revealed. Try another.');
                ask();
                return;
            }
            if (board[x][y] === 'B') {
                console.log('Boom! Game Over');
                rl.close();
                return;
            } else {
                display[x][y] = board[x][y];
                revealed++;
                if (revealed === totalToReveal) {
                    console.log('Congratulations! You win!');
                    showBoard();
                    rl.close();
                    return;
                }
                ask();
            }
        });
    }
    ask();
}

startGame();
